lib/templateFactory.js

```diff
@@ -6,6 +6,9 @@ const compareVersions = require('compare-versions');
 const schema = require('screwdriver-data-schema');
 let instance;
 
+const TEMPLATE_NAME_REGEX = schema.config.regex.FULL_TEMPLATE_NAME;
+const TEMPLATE_NAME_MATCH = 1;
```

Comment:
I think it makes sense to have them here. I can move it to the data-schema if that makes more sense. We have version regex matching values in this file as well.